PhysicsTools/TagAndProbe/plugins/AnythingToValueMap.h

```diff
@@ -0,0 +1,109 @@
+#ifndef PhysicsTools_TagAndProbe_interface_AnythingToValueMap_h
+#define PhysicsTools_TagAndProbe_interface_AnythingToValueMap_h
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Common/interface/View.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+namespace pat { namespace helper {
+
+  template<class Adaptor, class Collection = typename Adaptor::Collection, typename value_type = typename Adaptor::value_type> 
+  class AnythingToValueMap : public edm::EDProducer {
+    public:
+      typedef typename edm::ValueMap<value_type> Map;
+      typedef typename Map::Filler MapFiller;
+      explicit AnythingToValueMap(const edm::ParameterSet & iConfig) :
+            failSilently_(iConfig.getUntrackedParameter<bool>("failSilently", false)),
+            src_(iConfig.getParameter<edm::InputTag>("src")),
+            adaptor_(iConfig) { 
+                produces< Map >(adaptor_.label());
+            }
+      ~AnythingToValueMap() { }
+
+      virtual void produce(edm::Event & iEvent, const edm::EventSetup & iSetup) ;
+
+    private:
+      bool failSilently_;
+      edm::InputTag src_;
+      Adaptor adaptor_;
+  };
+
+template<class Adaptor, class Collection, typename value_type>
+void AnythingToValueMap<Adaptor,Collection,value_type>::produce(edm::Event & iEvent, const edm::EventSetup & iSetup) {
+    edm::Handle<Collection> handle;
+    iEvent.getByLabel(src_, handle);
```

Comment:
indeed, I have not ported it to consumes.
I suppose I can do and update the PR.

On Tue, Apr 21, 2015 at 5:46 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In PhysicsTools/TagAndProbe/plugins/AnythingToValueMap.h
> <https://github.com/cms-sw/cmssw/pull/8801#discussion_r28791447>:
>
> > +                produces< Map >(adaptor_.label());
> > +            }
> > +      ~AnythingToValueMap() { }
> > +
> > +      virtual void produce(edm::Event & iEvent, const edm::EventSetup & iSetup) ;
> > +
> > +    private:
> > +      bool failSilently_;
> > +      edm::InputTag src_;
> > +      Adaptor adaptor_;
> > +  };
> > +
> > +template<class Adaptor, class Collection, typename value_type>
> > +void AnythingToValueMap<Adaptor,Collection,value_type>::produce(edm::Event & iEvent, const edm::EventSetup & iSetup) {
> > +    edm::Handle<Collection> handle;
> > +    iEvent.getByLabel(src_, handle);
>
> no consumes?
> or is it supposed to happen in the Adaptor somewhere?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8801/files#r28791447>.
>